### ISSUE TITLE
Deprecation for Comparable `==`

### ISSUE THREAD

**Issue by bryanrite** [jdigger/git-process#138]
On ruby 2.2.0+, they changed the behaviour of Comparable#==

When running `git sync`, you now get:

```
$ git sync
Doing rebase-based sync
Rebasing branch against origin/master
/Users/user/.rvm/gems/ruby-2.2.2/gems/git-process-lib-2.0.4/lib/git-process/git_lib.rb:164: warning: Comparable#== will no more rescue exceptions of #<=> in the next release.
/Users/user/.rvm/gems/ruby-2.2.2/gems/git-process-lib-2.0.4/lib/git-process/git_lib.rb:164: warning: Return nil in #<=> if the comparison is inappropriate or avoid such comparison.
Fetching the latest changes from the server
Pushing to 'branch' on 'origin'.
```

**Comment by jdigger** [comment]
Thanks for the heads-up.

**Comment by jdigger** [comment]
This is fixed in v3.0.0
